Replace esbuild-plugin-polyfill-node with vite-plugin-node-polyfills

The plugin `esbuild-plugin-polyfill-node` is causing the following problems on some machines:

```
✘ [ERROR] The injected path "/path/to/project/node_modules/esbuild-plugin-polyfill-node/polyfills/__dirname.js" cannot be marked as external

✘ [ERROR] The injected path "/path/to/project/node_modules/esbuild-plugin-polyfill-node/polyfills/buffer.js" cannot be marked as external

✘ [ERROR] The injected path "/path/to/project/node_modules/esbuild-plugin-polyfill-node/polyfills/process.js" cannot be marked as external
```

This problem is reported [in this open issue](https://github.com/cyco130/esbuild-plugin-polyfill-node/issues/13) and one commenter suggests to replace `esbuild-plugin-polyfill-node` with `vite-plugin-node-polyfills`.

--- a/package.json
+++ b/package.json
@@ -87,7 +87,6 @@
     "@typescript-eslint/eslint-plugin": "^5.53.0",
     "@typescript-eslint/parser": "^5.53.0",
     "babel-preset-vite": "^1.1.3",
-    "esbuild-plugin-polyfill-node": "^0.3.0",
     "eslint": "^8.34.0",
     "eslint-plugin-react": "^7.32.2",
     "eslint-plugin-react-hooks": "^4.6.0",
@@ -98,6 +97,7 @@
     "ts-node": "^10.9.1",
     "typescript": "^5.3.3",
     "vite": "^5.3.5",
+    "vite-plugin-node-polyfills": "^0.22.0",
     "vitest": "^2.0.5"
   },
   "engines": {

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -1,11 +1,11 @@
 import preact from '@preact/preset-vite';
 import { defineConfig } from 'vite';
 
-import { polyfillNode } from 'esbuild-plugin-polyfill-node';
+import { nodePolyfills } from 'vite-plugin-node-polyfills';
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [preact()],
+  plugins: [preact(), nodePolyfills()],
   esbuild: {
     logOverride: { 'this-is-undefined-in-esm': 'silent' },
   },
@@ -29,11 +29,7 @@ export default defineConfig({
       supported: {
         bigint: true,
       },
-      plugins: [
-        // Enable esbuild polyfill plugins. We use the default settings.
-        // see https://github.com/cyco130/esbuild-plugin-polyfill-node?tab=readme-ov-file#options
-        polyfillNode({}),
-      ],
+      plugins: [],
     },
   },
 });

